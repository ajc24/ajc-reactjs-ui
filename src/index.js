export { default as Footer } from './ui-elements/Footer';
export { default as Header } from './ui-elements/Header';
export { default as HeaderLogo } from './ui-elements/HeaderLogo';
export { default as HeaderTitleAndSubtitle } from './ui-elements/HeaderTitleAndSubtitle';
export { default as MainContentPageTitle } from './ui-elements/MainContentPageTitle';
export { default as MainContentPanel } from './ui-elements/MainContentPanel';
export { default as MainContentText } from './ui-elements/MainContentText';
export { default as Menu } from './ui-elements/Menu';
export { default as Screenshot } from './ui-elements/Screenshot';
export { default as SmallMenuIcon } from './ui-elements/SmallMenuIcon';
export { default as SmallRightArrowIcon } from './ui-elements/SmallRightArrowIcon';
