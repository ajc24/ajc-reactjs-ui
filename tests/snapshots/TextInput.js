import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { TextInput } from '../../src';

describe('Text Input', () => {
  let componentDidMountSpy;

  beforeAll(() => {
    componentDidMountSpy = jest
      .spyOn(TextInput.prototype, 'componentDidMount')
      .mockImplementation(() => {});
  });

  afterAll(() => {
    componentDidMountSpy.mockRestore();
  });

  describe('Default props and rendering - No upper spacing, no character limit, default label text', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <TextInput id="default-text-input-id" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - With upper spacing and custom label text, no character limit', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <TextInput id="text-input-with-upper-spacing-id" labelText="With Upper Spacing and Custom Label:" addUpperSpacing={true} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - With character limit, no upper spacing or custom label text', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <TextInput id="text-input-with-character-limit-id" characterLimit={30} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
