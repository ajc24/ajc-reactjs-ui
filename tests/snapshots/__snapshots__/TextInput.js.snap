// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input Default props and rendering - No upper spacing, no character limit, default label text verifies the snapshot for the component 1`] = `
<div
  className="ajc-text-input-container ajc-text-black ajc-default"
>
  <label
    className="ajc-text-input-label"
    htmlFor="default-text-input-id"
  >
    Text Input Label Text:
  </label>
  <input
    className="ajc-text-input ajc-text-black ajc-default"
    id="default-text-input-id"
    name="default-text-input-id"
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`Text Input Transferred props and rendering - With character limit, no upper spacing or custom label text verifies the snapshot for the component 1`] = `
<div
  className="ajc-text-input-container ajc-text-black ajc-default"
>
  <label
    className="ajc-text-input-label"
    htmlFor="text-input-with-character-limit-id"
  >
    Text Input Label Text:
  </label>
  <input
    className="ajc-text-input ajc-text-black ajc-default"
    id="text-input-with-character-limit-id"
    name="text-input-with-character-limit-id"
    onChange={[Function]}
    type="text"
  />
  <span
    className="ajc-text-input-character-limit ajc-default ajc-text-black"
    id="text-input-with-character-limit-id-characters-remaining"
  />
</div>
`;

exports[`Text Input Transferred props and rendering - With upper spacing and custom label text, no character limit verifies the snapshot for the component 1`] = `
<div
  className="ajc-text-input-container ajc-text-black ajc-default"
>
  <label
    className="ajc-text-input-label ajc-text-input-label-upper-spacing"
    htmlFor="text-input-with-upper-spacing-id"
  >
    With Upper Spacing and Custom Label:
  </label>
  <input
    className="ajc-text-input ajc-text-black ajc-default"
    id="text-input-with-upper-spacing-id"
    name="text-input-with-upper-spacing-id"
    onChange={[Function]}
    type="text"
  />
</div>
`;
